src/main/webapp/app/shared/refresh/refresh-selector.component.ts

```diff
@@ -0,0 +1,112 @@
+import { Component, OnInit, OnDestroy, Input } from '@angular/core';
+import { Subscription } from 'rxjs/Subscription';
+
+import { Observable } from 'rxjs/Observable';
+import { JhiRefreshService } from './refresh.service';
+
+@Component({
+    selector: 'jhi-refresh-selector',
+    templateUrl: './refresh-selector.component.html',
+    styleUrls: [
+        'refresh-selector.component.scss'
+    ]
+})
+export class JhiRefreshSelectorComponent implements OnInit, OnDestroy {
+
+    activeRefreshTime: number;
+    refreshTimes: number[];
+    refreshTimer: Subscription;
+    refreshChangedSubscription: Subscription;
+
+    /** True if dropdown-menu must be open in left direction (right direction is by default) **/
+    @Input()
+    toTheLeft: boolean = false;
```

Comment:
After reflection, it's true that an input on directives is more related to a behavior rather than a css, so indeed, for this case, we only have to add the class on the directive.